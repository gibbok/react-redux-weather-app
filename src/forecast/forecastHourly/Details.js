import React from 'react'
// import Detail from './Detail'
import DetailDayGroup from './DetailDayGroup'

/*
 {
  date1: [],
  date2: []
 }
*/

const Details = ({ forecastHourly }) => {
  let result = forecastHourly.reduce((acc, item) => {
    (acc[item.momentId] || (acc[item.momentId] = [])).push(item)
    return acc
  }, {})
  console.log(result)
  return (
    <div>
      {forecastHourly.map((day, index) =>
        <DetailDayGroup
          key={index}
          day={day}
      />
    )}
    </div>
  )
}

// const Details = ({ forecastHourly }) => (
//   <div>
//     <ul>
//       {forecastHourly.map((day, index) =>
//         <Detail
//           key={index}
//           {...forecastHourly[index]}
//       />
//     )}
//     </ul>
//   </div>
// )

export default Details
