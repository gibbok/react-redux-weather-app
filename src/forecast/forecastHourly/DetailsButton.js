// @flow
import * as React from 'react'
import {Button} from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +onDetailsClick: Function,
}
/* eslint-enable no-undef */

const DetailsButton:React.StatelessFunctionalComponent<any> = ({ onDetailsClick }:PropsType):React.Element<any> => {
  return (
    <Button onClick={(e) => onDetailsClick(e)}>Details</Button>
  )
}

export default DetailsButton
