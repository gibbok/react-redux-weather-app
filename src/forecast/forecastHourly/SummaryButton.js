// @flow
import * as React from 'react'
import {FlatButton} from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +onSummaryClick: Function
}
/* eslint-enable no-undef */

const SummaryButton:React.StatelessFunctionalComponent<any> = ({ onSummaryClick }:PropsType):React.Element<any> => {
  return (
    <FlatButton onClick={(e) => onSummaryClick(e)}>Summary</FlatButton>
  )
}

export default SummaryButton
