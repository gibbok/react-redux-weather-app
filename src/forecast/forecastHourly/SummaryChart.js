import React from 'react'
import { LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip } from 'recharts'
import SummaryChartIcon from './SummaryChartIcon'
import moment from 'moment'

const SummaryChart = ({ forecastHourly }) => {
  let forecastHourlyFormat = forecastHourly.map(data => {
    return {
      momentId: data.momentId,
      timespan: data.timespan,
      temp: Math.round(data.temp),
      weatherIconCode: data.weatherIconCode,
      rain: data.rain,
      hour: moment.unix(data.timespan).format('hh A')
    }
  })
  return (
    <LineChart width={800} height={500} data={forecastHourlyFormat} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
      <Line type='monotone' dataKey='temp' stroke='#8884d8' dot={<SummaryChartIcon />} />
      <CartesianGrid stroke='#ccc' strokeDasharray='5 5' />
      <XAxis dataKey='hour' />
      <YAxis dataKey='temp' />
      <Tooltip />
    </LineChart>
  )
}

// const SummaryChart = ({ forecastHourly }) => (
//   <LineChart width={800} height={500} data={forecastHourly} margin={{ top: 20, right: 20, bottom: 20, left: 20 }}>
//     <Line type='monotone' dataKey='temp' stroke='#8884d8' dot={<SummaryChartIcon />} />
//     <CartesianGrid stroke='#ccc' strokeDasharray='5 5' />
//     <XAxis dataKey='hour' />
//     <Tooltip />
//   </LineChart>
// )

export default SummaryChart
