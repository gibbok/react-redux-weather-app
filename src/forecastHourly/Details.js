import React from 'react'
import Detail from './Detail'

const Details = ({ forecastHourly }) => (
  <div>
    <ul>
      {forecastHourly.map((day, index) =>
        <Detail
          key={index}
          {...forecastHourly[index]}
      />
    )}
    </ul>

  </div>
)

export default Details
