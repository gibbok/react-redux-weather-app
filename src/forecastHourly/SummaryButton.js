import React from 'react'

const SummaryButton = ({ onSummaryClick }) => {
  return (
    <button type='button' onClick={(e) => onSummaryClick(e)}>Summary</button>
  )
}

export default SummaryButton
