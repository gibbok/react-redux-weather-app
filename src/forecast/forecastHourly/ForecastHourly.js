// @flow
import * as React from 'react'
import {ForecastHourlyType} from '../../types' // eslint-disable-line no-unused-vars
import SummaryChart from './SummaryChart'
import Details from './Details'
import SummaryButton from './SummaryButton'
import DetailsButton from './DetailsButton'
import { withStyles, Typography } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +forecastHourly:ForecastHourlyType,
  +activeTypeReport: string,
  +onSummaryClick: Function,
  +onDetailsClick: Function,
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  const spaceUnit = theme.spacing.unit
  return ({
    root: {
      width: 900
    },
    title: {
      marginBottom: spaceUnit
    }
  })
}

const ForecastHourly:React.StatelessFunctionalComponent<any> = ({ forecastHourly, activeTypeReport, onSummaryClick, onDetailsClick, classes }:PropsType):React.Element<any> => {
  return (
    <div className={classes.root}>
      <div className={classes.title}>
        <Typography type='headline'>Hourly</Typography>
      </div>
      <SummaryButton onSummaryClick={onSummaryClick} />
      <DetailsButton onDetailsClick={onDetailsClick} />
      {activeTypeReport === 'summary'
      ? <SummaryChart forecastHourly={forecastHourly} />
      : <Details forecastHourly={forecastHourly} />}
    </div>
  )
}

export default withStyles(styles)(ForecastHourly)
