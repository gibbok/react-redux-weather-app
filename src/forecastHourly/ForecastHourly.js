import React from 'react'
import SummaryChart from './SummaryChart'
import Details from './Details'
import SummaryButton from './SummaryButton'
import DetailsButton from './DetailsButton'

const ForecastHourly = ({ forecastHourly, activeTypeReport, onSummaryClick, onDetailsClick }) => {
  const isReportSummary = activeTypeReport === 'summary'
  const isReportDetails = activeTypeReport === 'details'
  return (
    <div>
      <SummaryButton onSummaryClick={onSummaryClick} />
      <DetailsButton onDetailsClick={onDetailsClick} />
      {isReportSummary ? (<SummaryChart forecastHourly={forecastHourly} />) : null}
      {isReportDetails ? (<Details forecastHourly={forecastHourly} />) : null}
    </div>
  )
}

export default ForecastHourly
