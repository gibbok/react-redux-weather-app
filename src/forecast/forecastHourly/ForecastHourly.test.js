import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import ForecastHourly from './ForecastHourly'
import SummaryButton from './SummaryButton'
import DetailsButton from './DetailsButton'
import Details from './Details'
import SummaryChart from './SummaryChart'

describe('<ForecastHourly />', () => {
  it('should render', () => {
    const wrapper = shallow(<ForecastHourly activeTypeReport={'summary'} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should contain SummaryButton', () => {
    const wrapper = shallow(<ForecastHourly />)
    expect(wrapper.find(SummaryButton).length).toBe(1)
    expect(wrapper.find(DetailsButton).length).toBe(1)
    expect(wrapper.find(Details).length).toBe(1)
  })

  it('should render SummaryChart when activeTypeReport is summary', () => {
    const wrapper = shallow(<ForecastHourly activeTypeReport={'summary'} />)
    expect(wrapper.find(SummaryChart).length).toBe(1)
  })
  it('should render Details when activeTypeReport is not summary', () => {
    const wrapper = shallow(<ForecastHourly />)
    expect(wrapper.find(Details).length).toBe(1)
  })
})
