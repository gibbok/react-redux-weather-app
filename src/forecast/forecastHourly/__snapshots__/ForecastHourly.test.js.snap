// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastHourly /> should render 1`] = `
<div
  className="ForecastHourly-root-1"
>
  <div
    className="ForecastHourly-title-2"
  >
    <withStyles(Typography)
      type="headline"
    >
      Hourly
    </withStyles(Typography)>
  </div>
  <SummaryButton />
  <DetailsButton />
  <SummaryChart />
</div>
`;
